### PR TITLE
[WIP] Makes overload respect override immunity and broken +  potentially adding to the blacklist

### PR DESCRIPTION
**Pull request by Supermichael777**
:cl:
balance: Machine overload now has a blacklist shared with override. Both can no longer act on broken machines.
fix: The override blacklisting now actually works.
/:cl:

Currently machine overload and machine override can target every type of obj/machine bar one: the cover of the portable turret. Overload can even target that. Now this might sound like working as intended but a long list of things are machines, including rather ubiquitous pipes and holo-pads. This pr will act as a list of things it shouldn't be able to blow up by community suggestion. I'm almost certainly going to add transit tubing and most of the atmos/disposals equipment but I want some feedback on what else should go in as well as HEALTHY DEBATE about whether each suggestion is freeze friendly.

Things that logically should not explode + community input:

- [x] anything with status BROKEN

- [x ] atmos pipes and manual equipment

- [ ] disposals pipes

- [ ] transit tubing

for your consideration
 
1. Every kind of atmos machine (vents pumps, filters, etc.)
2. Disposals machinery (the kind that goes on the terminals)
3. Suggestions

### CODE REVIEW

**Comment by Shadowlight213** [comment]
Never forget when the singularity was a machine and you could override it.

**Comment by Tacolizard** [comment]
Holopads should totally be explodeable. They're machinery made specifically to be AI controllable

**Comment by Repukan** [comment]
Holopads should remain explode-able.

Also I feel this would be better as a forum post as well.

**Comment by Supermichael777** [comment]
Why the fuck is this a proc? that's trash!

File: code/game/gamemodes/malfunction/Malf_Modules.dm [hunk 1]
```diff
@@ -322,6 +330,10 @@
 	if (istype(M, /obj/machinery))
 		if(!M.can_be_overridden())
 			to_chat(src, "Can't override this device.")
+			return
+		if(stat & BROKEN)
```
Comment (Cyberboss) [inline code/game/gamemodes/malfunction/Malf_Modules.dm]: Should probably use is_operational()

File: code/modules/atmospherics/machinery/atmosmachinery.dm [hunk 2]
```diff
@@ -23,7 +23,8 @@ Pipelines + Other Objects -> Pipe network
 	var/can_unwrench = 0
 	var/initialize_directions = 0
 	var/pipe_color
-
+	var/can_be_overridden = 0
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/atmosmachinery.dm]: FALSE

File: code/modules/atmospherics/machinery/atmosmachinery.dm [hunk 3]
```diff
@@ -330,3 +331,9 @@ Pipelines + Other Objects -> Pipe network
 	else
 		// atmosinit() calls update_icon(), so we don't need to call it
 		update_icon()
+		
+/obj/machinery/atmospherics/can_be_overridden()
+	if(can_be_overridden)
+		return 1
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/atmosmachinery.dm]: TRUE

File: code/modules/atmospherics/machinery/atmosmachinery.dm [hunk 4]
```diff
@@ -330,3 +331,9 @@ Pipelines + Other Objects -> Pipe network
 	else
 		// atmosinit() calls update_icon(), so we don't need to call it
 		update_icon()
+		
+/obj/machinery/atmospherics/can_be_overridden()
+	if(can_be_overridden)
+		return 1
+	else
+		return 0
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/atmosmachinery.dm]: FALSE

File: code/modules/atmospherics/machinery/components/binary_devices/passive_gate.dm [hunk 5]
```diff
@@ -13,6 +13,7 @@ Passive gate is similar to the regular pump except:
 	desc = "A one-way air valve that does not require power."
 
 	can_unwrench = 1
+	can_be_overridden = 0
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/components/binary_devices/passive_gate.dm]: FALSE

File: code/modules/atmospherics/machinery/components/binary_devices/valve.dm [hunk 6]
```diff
@@ -6,8 +6,9 @@ It's like a regular ol' straight pipe, but you can turn it on and off.
 	icon_state = "mvalve_map"
 	name = "manual valve"
 	desc = "A pipe valve"
-
+	
 	can_unwrench = 1
+	can_be_overridden = 0
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/components/binary_devices/valve.dm]: FALSE

File: code/modules/atmospherics/machinery/components/binary_devices/valve.dm [hunk 7]
```diff
@@ -60,6 +61,7 @@ It's like a regular ol' straight pipe, but you can turn it on and off.
 	desc = "A digitally controlled valve."
 	icon_state = "dvalve_map"
 	valve_type = "d"
+	can_be_overridden = 1
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/components/binary_devices/valve.dm]: TRUE

File: code/modules/atmospherics/machinery/components/components_base.dm [hunk 8]
```diff
@@ -6,6 +6,7 @@ On top of that, now people can add component-speciic procs/vars if they want!
 /obj/machinery/atmospherics/components
 	var/welded = 0 //Used on pumps and scrubbers
 	var/showpipe = 0
+	can_be_overridden = 1
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/components/components_base.dm]: TRUE

File: code/modules/atmospherics/machinery/components/unary_devices/heat_exchanger.dm [hunk 9]
```diff
@@ -6,6 +6,7 @@
 	desc = "Exchanges heat between two input gases. Setup for fast heat transfer"
 
 	can_unwrench = 1
+	can_be_overridden = 0
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/components/unary_devices/heat_exchanger.dm]: FALSE

File: code/modules/atmospherics/machinery/components/unary_devices/portables_connector.dm [hunk 10]
```diff
@@ -7,6 +7,7 @@
 	var/obj/machinery/portable_atmospherics/connected_device
 	use_power = 0
 	level = 0
+	can_be_overridden = 0
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/components/unary_devices/portables_connector.dm]: FALSE

File: code/modules/atmospherics/machinery/components/unary_devices/tank.dm [hunk 11]
```diff
@@ -9,6 +9,7 @@
 	var/volume = 10000 //in liters, 1 meters by 1 meters by 2 meters
 	density = 1
 	var/gas_type = 0
+	can_be_overridden = 0
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/components/unary_devices/tank.dm]: FALSE

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/malfunction/Malf_Modules.dm [hunk 12]
```diff
@@ -288,6 +288,14 @@
 
 	if(!canUseTopic())
 		return
+	
+	if(!M.can_be_overridden())
```
Comment (lzimann) [inline code/game/gamemodes/malfunction/Malf_Modules.dm]: Why is this can_be_overriden a proc?

**Comment by Supermichael777** [comment]
i have lost all drive to make this better

**Pull request closed without merging**